src/main/java/ml/shifu/shifu/util/updater/BasicUpdater.java

```diff
@@ -231,8 +231,7 @@ private void addSegColumns(Set<NSColumn> segColumnSet, String columnName) {
         // Only do it when segment exists.
         if (this.isForSegs) {
             ColumnConfig originalColumnConfig = columnConfigMap.get(columnName);
-            // We need make sure the original column exist and the bound is within [0, originalColumnAmount).
-            if (originalColumnConfig == null && originalColumnConfig.getColumnNum() < originalColumnAmount) {
+            if (originalColumnConfig == null) {
```

Comment:
We don't need to check the column amount because we checked similar condition later `if (segColumnConfigNum >= columnConfigList.size()) {`.